# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmInstance.h
# Commit: 67ae30727ed8
# Full Hash: 67ae30727ed835cf73153a4ee8f0c9c7931aab09
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2021-12-23 09:34:36
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1642412 - wasm: Remove unnecessary code from Instance for inline table methods. r=lth
#   
#   This commit removes support for anyref in Instance::tableGet/tableSet,
#   making it clear that these functions are only used for function tables.
#   Instance::tableSize is removed as no longer needed.
# ==============================================================================

diff -r 90fa382b5a20 -r 67ae30727ed8 js/src/wasm/WasmInstance.h
--- a/js/src/wasm/WasmInstance.h	Wed Dec 22 17:36:30 2021 +0000
+++ b/js/src/wasm/WasmInstance.h	Wed Dec 22 17:36:30 2021 +0000
@@ -254,13 +254,12 @@
                            uint32_t dstTableIndex, uint32_t srcTableIndex);
   static int32_t tableFill(Instance* instance, uint32_t start, void* value,
                            uint32_t len, uint32_t tableIndex);
-  static void* tableGet(Instance* instance, uint32_t index,
-                        uint32_t tableIndex);
+  static void* tableGetFunc(Instance* instance, uint32_t index,
+                            uint32_t tableIndex);
   static uint32_t tableGrow(Instance* instance, void* initValue, uint32_t delta,
                             uint32_t tableIndex);
-  static int32_t tableSet(Instance* instance, uint32_t index, void* value,
-                          uint32_t tableIndex);
-  static uint32_t tableSize(Instance* instance, uint32_t tableIndex);
+  static int32_t tableSetFunc(Instance* instance, uint32_t index, void* value,
+                              uint32_t tableIndex);
   static int32_t tableInit(Instance* instance, uint32_t dstOffset,
                            uint32_t srcOffset, uint32_t len, uint32_t segIndex,
                            uint32_t tableIndex);